paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -343,9 +343,12 @@ def downscale_aws_resource(self, filtered_slaves, current_capacity, target_capac
             current_capacity = new_capacity
             mesos_state = get_mesos_master().state_summary()
             if filtered_sorted_slaves:
-                filtered_slaves = get_mesos_task_count_by_slave(mesos_state, slaves_list=filtered_sorted_slaves)
-            else:
-                filtered_slaves = filtered_sorted_slaves
+                task_counts = get_mesos_task_count_by_slave(mesos_state,
+                                                            slaves_list=[{'task_counts': slave.task_counts}
```

Comment:
yeah, this is really nasty. You're kind of abusing the fact that {{get_mesos_tsak_count_by_slave}} doesn't do any type checking, plus that function is just nasty.